src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java

```diff
@@ -534,5 +538,12 @@ static String getResource(String key) {
             {"R_unknownUTF8SupportValue", "Unknown value for UTF8 support."},
             {"R_illegalWKT", "Illegal Well-Known text. Please make sure Well-Known text is valid."},
             {"R_illegalTypeForGeometry", "{0} is not supported for Geometry."},
-            {"R_illegalWKTposition", "Illegal character in Well-Known text at position {0}."},};
+            {"R_illegalWKTposition", "Illegal character in Well-Known text at position {0}."},
+            {"R_ADALMissing", "Failed to load ADAL4J Java library for performing {0} authentication."},
+            {"R_DLLandADALMissing",
+                    "Failed to load both sqljdbc_auth.dll and ADAL4J Java library for performing {0} authentication. Please install one of them to proceed."},
+            {"R_MSITokenFailureClientId", "Couldn't acquire access token from IMDS, verify your clientId."},
```

Comment:
Please make sure the error messages are consistent and also include MSI. Also, do you think we should be using abbreviation MSI or full term Managed Service Identity in the error message?